src/material-examples/table-overview/table-overview-example.ts

```diff
@@ -27,8 +30,8 @@ export class TableOverviewExample implements OnInit {
   displayedColumns: string[] = ['id', 'name', 'progress', 'color'];
   dataSource: MatTableDataSource<UserData>;
 
-  @ViewChild(MatPaginator) paginator: MatPaginator;
-  @ViewChild(MatSort) sort: MatSort;
+  @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
```

Comment:
Couldn't we move the usages to `ngAfterViewInit`, rather than marking them as `static`? Since it's an example it won't be a breaking change.

Reply:
The issue with the table is that it needs the paginator and sort to figure out the initially rendered rows before initialization. If we wait until after view init, the table will initially render all rows.

Reply:
I see, although it's a little unfortunate that we have to require people to use static queries.